### ISSUE TITLE
WordPress 4.3 Memory Leak?

### ISSUE THREAD

**Issue by vicatcu** [woothemes/woocommerce#8931]
Hey gang, I updated Wordpress to 4.3 today and Woocommerce to 2.4.5 (and ran the data updater fine), and the admin dashboard continued to work fine. But when I went to check on the shop itself, it had crashed badly with the following PHP error.

> Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 130968 bytes) in /opt/bitnami/apps/wordpress/htdocs/wp-includes/formatting.php on line 1456

... which is this innocuous line (call to remove_accents) ...

```php
 function sanitize_title( $title, $fallback_title = '', $context = 'save' ) {
        $raw_title = $title;
        if ( 'save' == $context )
                $title = remove_accents($title);
```
I went into my php.ini and bumped memory_limit up from 128M to 512M and the page at least loads without that PHP error, but my EC2 T2.small instance is reporting pretty much constant 100% utilization now, even when I pushed the memory_limit up to 1G. 

I'm really hoping this is a transient problem, can you shed any light on it? My website shop running smoothly is pretty important to me :-). I don't feel confident in my ability to "go back" to before the updates, and at any rate that's at best a bandaid. Any help you can offer would be greatly appreciated!

**Comment by mikejolley** [comment]
Frontend? How many posts per page? Do all pages shown the same symptoms, i.e. do certain types of products like variable cause it rather than simple?

I think the sanitize_title is a red herring - just being the last thing called before memory ran out.

**Comment by vicatcu** [comment]
Happy to debug and provide more data with guidance. I'm experienced with
browser Developer Tools and in the linux shell. Seems to me very likely
*not* a front end issue since I am getting eventual 503 Service Unavailable
responses when page load lags, and high server load with very minimal page
traffic. This happens simply loading the index of the site, not any product
listing in particular.

**Comment by vicatcu** [comment]
@mikejolley fwiw, I just counted 69 requests on initial page load using Firebug on Firefox. I can't get the homepage to load at all using Chrome at the moment.

**Comment by mikejolley** [comment]
Where is the site?

**Comment by vicatcu** [comment]
Hi Mike, it's on a AWS T2.small instance. The URL is shop.wickeddevice.com.

Victor Aprea // Wicked Device
victor.aprea@wickeddevice.com
T: @vicatcu <http://twitter.com/vicatcu/>

**Comment by mikejolley** [comment]
It was loading normal (https://dl.dropboxusercontent.com/s/8qhpyuo66yp9t65/2015-08-24%20at%2015.35.png?dl=0) until I added an item to the cart when it died.

Post your system status report.

**Comment by vicatcu** [comment]
### WordPress Environment ###

Home URL: http://shop.wickeddevice.com
Site URL: http://shop.wickeddevice.com
WC Version: 2.4.5
Log Directory Writable: ✔  /opt/bitnami/apps/wordpress/htdocs/wp-content/uploads/wc-logs/
WP Version: 4.3
WP Multisite: –
WP Memory Limit: 96 MB
WP Debug Mode: –
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 5.4.22
PHP Post Max Size: 40 MB
PHP Time Limit: 120
PHP Max Input Vars: 1000
SUHOSIN Installed: –
MySQL Version: 5.5.34
Max Upload Size: 40 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.4.5
: 
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_termmeta: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔

### Active Plugins (23) ###

Cache Images: by Matt Mullenweg – 3.1
Disable Comments: by Samir Shah – 1.3.2
Google XML Sitemaps: by Arne Brachhold – 4.0.8
Menu Exporter: by Joe Hoyle - Human Made Limited – 1
Safe Redirect Manager: by Taylor Lovett (10up) – 1.7.7
SyntaxHighlighter Evolved: by Alex Mills (Viper007Bond) – 3.1.13
TablePress: by Tobias Bäthge – 1.6.1
Wicked Device Inventory System Hooks: by Victor Aprea – 1.0.0
WooCommerce Bulk Pricing: by WP Lab – 1.8.1
WooCommerce Export Customer Email: by MH Mithu – 1.3
WooCommerce Stripe Gateway: by Mike Jolley – 2.5.4
WooCommerce MailChimp: by Adam Anderly – 1.3.6
WooCommerce Product Bundles: by SomewhereWarm – 3.6.4 – 4.11.1 is available
WooCommerce Product CSV Import Suite: by Mike Jolley – 1.8.3 – 1.10.1 is available
WooCommerce Predictive Search PRO: by A3 Revolution – 3.0.0
WooCommerce USPS Shipping: by WooThemes – 4.2.12
WooCommerce: by WooThemes – 2.4.5
WooThemes Helper: by WooThemes – 1.5.9
WordPress FAQ Manager: by Andrew Norcross – 1.331
WordPress Importer: by wordpressdotorg – 0.6.1
WP-DraftsForFriends: by Lester 'GaMerZ' Chan – 1.0.2
WP-Mail-SMTP: by Callum Macdonald – 0.9.5
Zopim Widget: by Zopim – 1.3.9

### Settings ###

Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2

### API ###

API Enabled: ✔
API Version: 3.0.0

### WC Pages ###

Shop Base: #503 - /shop/
Cart: #604 - /cart/
Checkout: #505 - /checkout/
My Account: #7 - /my-account/

### Taxonomies ###

Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)


### Theme ###

Name: WooStore
Version: 1.8.1
Author URL: http://www.woothemes.com
Child Theme: ✕ – If you're modifying WooCommerce or a parent theme you didn't build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

**Comment by mikejolley** [comment]
Turn off anything to do with bulk pricing or redirects.

**Comment by WickedDevice** [comment]
OK, I've deactivated "Safe Redirect Manager" and "WooCommerce Bulk Pricing". System Status now looks like this:


### WordPress Environment ###

Home URL: http://shop.wickeddevice.com
Site URL: http://shop.wickeddevice.com
WC Version: 2.4.5
Log Directory Writable: ✔  /opt/bitnami/apps/wordpress/htdocs/wp-content/uploads/wc-logs/
WP Version: 4.3
WP Multisite: –
WP Memory Limit: 96 MB
WP Debug Mode: –
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 5.4.22
PHP Post Max Size: 40 MB
PHP Time Limit: 120
PHP Max Input Vars: 1000
SUHOSIN Installed: –
MySQL Version: 5.5.34
Max Upload Size: 40 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.4.5
: 
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_termmeta: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔

### Active Plugins (21) ###

Cache Images: by Matt Mullenweg – 3.1
Disable Comments: by Samir Shah – 1.3.2
Google XML Sitemaps: by Arne Brachhold – 4.0.8
Menu Exporter: by Joe Hoyle - Human Made Limited – 1
SyntaxHighlighter Evolved: by Alex Mills (Viper007Bond) – 3.1.13
TablePress: by Tobias Bäthge – 1.6.1
Wicked Device Inventory System Hooks: by Victor Aprea – 1.0.0
WooCommerce Export Customer Email: by MH Mithu – 1.3
WooCommerce Stripe Gateway: by Mike Jolley – 2.5.4
WooCommerce MailChimp: by Adam Anderly – 1.3.6
WooCommerce Product Bundles: by SomewhereWarm – 3.6.4 – 4.11.1 is available
WooCommerce Product CSV Import Suite: by Mike Jolley – 1.8.3 – 1.10.1 is available
WooCommerce Predictive Search PRO: by A3 Revolution – 3.0.0
WooCommerce USPS Shipping: by WooThemes – 4.2.12
WooCommerce: by WooThemes – 2.4.5
WooThemes Helper: by WooThemes – 1.5.9
WordPress FAQ Manager: by Andrew Norcross – 1.331
WordPress Importer: by wordpressdotorg – 0.6.1
WP-DraftsForFriends: by Lester 'GaMerZ' Chan – 1.0.2
WP-Mail-SMTP: by Callum Macdonald – 0.9.5
Zopim Widget: by Zopim – 1.3.9

### Settings ###

Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2

### API ###

API Enabled: ✔
API Version: 3.0.0

### WC Pages ###

Shop Base: #503 - /shop/
Cart: #604 - /cart/
Checkout: #505 - /checkout/
My Account: #7 - /my-account/

### Taxonomies ###

Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)


### Theme ###

Name: WooStore
Version: 1.8.1
Author URL: http://www.woothemes.com
Child Theme: ✕ – If you're modifying WooCommerce or a parent theme you didn't build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

**Comment by vicatcu** [comment]
@mikejolley what should I do/try next?

**Comment by vicatcu** [comment]
Still getting random 503 errors and high server load

**Comment by mikejolley** [comment]
@kloon any advice here?

**Comment by kloon** [comment]
Do you get the same results when switching your theme to 2014?

**Comment by vicatcu** [comment]
Sure I can try that... in the interest of keeping ya'll posted, I updated the WooStore Theme and I updated  the WooCommerce plugin just now. Here's the latest System Status:


### WordPress Environment ###

Home URL: http://shop.wickeddevice.com
Site URL: http://shop.wickeddevice.com
WC Version: 2.4.6
Log Directory Writable: ✔  /opt/bitnami/apps/wordpress/htdocs/wp-content/uploads/wc-logs/
WP Version: 4.3
WP Multisite: –
WP Memory Limit: 96 MB
WP Debug Mode: –
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 5.4.22
PHP Post Max Size: 40 MB
PHP Time Limit: 120
PHP Max Input Vars: 1000
SUHOSIN Installed: –
MySQL Version: 5.5.34
Max Upload Size: 40 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.4.6
: 
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_termmeta: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔

### Active Plugins (21) ###

Cache Images: by Matt Mullenweg – 3.1
Disable Comments: by Samir Shah – 1.3.2
Google XML Sitemaps: by Arne Brachhold – 4.0.8
Menu Exporter: by Joe Hoyle - Human Made Limited – 1
SyntaxHighlighter Evolved: by Alex Mills (Viper007Bond) – 3.1.13
TablePress: by Tobias Bäthge – 1.6.1
Wicked Device Inventory System Hooks: by Victor Aprea – 1.0.0
WooCommerce Export Customer Email: by MH Mithu – 1.3
WooCommerce Stripe Gateway: by Mike Jolley – 2.5.4
WooCommerce MailChimp: by Adam Anderly – 1.3.6
WooCommerce Product Bundles: by SomewhereWarm – 3.6.4 – 4.11.1 is available
WooCommerce Product CSV Import Suite: by Mike Jolley – 1.8.3 – 1.10.2 is available
WooCommerce Predictive Search PRO: by A3 Revolution – 3.0.1
WooCommerce USPS Shipping: by WooThemes – 4.2.12
WooCommerce: by WooThemes – 2.4.6
WooThemes Helper: by WooThemes – 1.5.9
WordPress FAQ Manager: by Andrew Norcross – 1.331
WordPress Importer: by wordpressdotorg – 0.6.1
WP-DraftsForFriends: by Lester 'GaMerZ' Chan – 1.0.2
WP-Mail-SMTP: by Callum Macdonald – 0.9.5
Zopim Widget: by Zopim – 1.3.9

### Settings ###

Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2

### API ###

API Enabled: ✔
API Version: 3.0.0

### WC Pages ###

Shop Base: #503 - /shop/
Cart: #604 - /cart/
Checkout: #505 - /checkout/
My Account: #7 - /my-account/

### Taxonomies ###

Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)


### Theme ###

Name: WooStore
Version: 1.8.3
Author URL: http://www.woothemes.com
Child Theme: ✕ – If you're modifying WooCommerce or a parent theme you didn't build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

**Comment by vicatcu** [comment]
I also turned the PHP memory_limit back down to 128M instead of 1G so as to stop masking "the real problem", if you visit the site right now you'll see 

    Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 130968 bytes) in /opt/bitnami/apps/wordpress/htdocs/wp-includes/formatting.php on line 1456

**Comment by vicatcu** [comment]
@kloon  @mikejolley Applying the 2014 theme results in the exact same Fatal error on loading the index.

**Comment by vicatcu** [comment]
Site loads when I deactivate every plugin.... I'll start adding them back one at a time.

**Comment by vicatcu** [comment]
So I think it's WooCommerce Product Bundles: by SomewhereWarm – 3.6.4 – 4.11.1 is available, which makes some sense given that it hasn't been updated in quite some time...

**Comment by vicatcu** [comment]
So I've re-enabled all the Plugsin I think I needed from what was previously activated, except for the WooCommerce Product Bundles. You can probably close this issue at this point. As a quick aside, having previously created product bundles, can you summarize what is the impact of not having that plugin activated is?

**Comment by mikejolley** [comment]
Disabling it would only stop bundled product type products working.

If there is a bundles issue unresolved by updates, log a ticket on support.woothemes.com

Thanks
